### PR TITLE
Refactor kubernetes roles

### PR DESCRIPTION
**Pull request by ingvagabund**
Continuing effort [1] of cleaning roles for kubernetes, master and node roles.

TODO:
- [x] clean `kubernetes` role
- [x] move `download_bins.yml` under `node` and `master` and enable installation of downloaded binaries for Fedora, CentOS and other OSes.
- [x] extend the installation options with installation of distribution rpm (not just the latest)
- [x] clean and refactor `node` role
- [x] clean and refactor `master` role
- [x] introduce `deploy-master.yml` and `deploy-node.yml` playbooks, include them in the top level playbook
- [x] tag relevant tasks and introduce new scripts
- [x] move CPU and Memory accounting under node role (inside `kubelet-configuration.yml`)

Some items of the TODO (new playbooks, new scripts) can be resolved only once the https://github.com/kubernetes/contrib/pull/967 is merged.

[1] https://github.com/kubernetes/contrib/issues/1298

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1444)

### CODE REVIEW

**Comment by ingvagabund** [comment]
@stephenrlouie @adamschaub @danehans @cgwalters @eparis

File: ansible/roles/kubernetes/tasks/main.yml [hunk 1]
```diff
@@ -3,21 +3,8 @@
...
   when: ansible_distribution == "Fedora" and ansible_distribution_major_version == "20"
 
-- name: Update {{ kube_script_dir }} if this is atomic
-  set_fact:
-    kube_script_dir: "/usr/local/libexec/kubernetes"
-  when: is_atomic and kube_script_dir == "/usr/libexec/kubernetes"
-
-- name: Initialize the python_bin fact
-  set_fact:
-    python_bin: "python"
```
Comment (ingvagabund) [inline ansible/roles/kubernetes/tasks/main.yml]: This is a candidate for moving to ``default/main.yml``.

**Comment by ingvagabund** [comment]
@jasonbrooks how can I check the ansible playbook deploys AH correctly?

**Comment by jasonbrooks** [comment]
@ingvagabund I usually use vagrant:

```
cd contrib/ansible/vagrant
export OS_IMAGE=centosatomic
vagrant up
```

I just tried it w/ your branch, and it seems to work.

**Comment by ingvagabund** [comment]
@gouyang @mdshuai how often do you use ansible to deploy the cluster? Can you take a look at the PR and see if there is anything else missing in your use case(s)?

**Comment by mdshuai** [comment]
@ingvagabund I usually use ansible to deploy the cluster for our user story test on upstream. Sometime need to use the latest code; I think in this pr 'localBuild' and 'github-release' can help my problem. When this complete, I will integrate ansible with our qe jenkins job to setup k8s cluster automatically.
This pr really useful. thanks for your work.

**Comment by gouyang** [comment]
@ingvagabund I use ansible to deploy the cluster daily. Usually I will deploy the cluster via ansible, clean up the cluster after some testing and deploy it again, so it's very often.

I've tested this PR, it can run smoothly. But I got some problems:
1. There is a tag "binary-update" for master and node, I could use them to skip installation for them, but not for etcd,docker,flannel and others like easy-rsa, how to skip installation for the laters?

2. Regarding file deploy-cluster.yml, the role order can be improved.
-  Need to move `flannel` above `docker`, there is a problem when execute `docker` before `flannel` because docker service depends on flannel, if the flannel configure file is outdated and the service could not be started correctly, the whole playbook failed and exit. To execute flannel before docker, the configure files will be corrected and no errors happen. 
- `node` can be moved above `kubernetes-addons` because addons need node to be ready to deploy.

These are what I got so far, please let me know if any issue need to open for the problems.

**Comment by ingvagabund** [comment]
> 1. There is a tag "binary-update" for master and node, I could use them to skip installation for them, but not for etcd,docker,flannel and others like easy-rsa, how to skip installation for the laters?

That is not yet implemented. I plan to update all roles by introducing ``install`` tag.

> 2. Regarding file deploy-cluster.yml, the role order can be improved.
> 
>   - Need to move `flannel` above `docker`, there is a problem when execute `docker` before `flannel` because docker service depends on flannel, if the flannel configure file is outdated and the service could not be started correctly, the whole playbook failed and exit. To execute flannel before docker, the configure files will be corrected and no errors happen. 

Actually, it is the opposite. Docker needs to be installed first as flannel reuses/runs over docker's network. If there is a problem after re-running the playbook, please open an issue with steps how to reproduce it, with logs, etc.

>   - `node` can be moved above `kubernetes-addons` because addons need node to be ready to deploy.

The is reasonable. Not sure if the ``kubernetes-addons`` role checks if all addons are running. From my observations it seems it does not. So even if any of the nodes is not ready, it will get ready eventually. Once done, addons gets scheduled and running. Still, I agree with that. Can you open an issue for that?

**Comment by gouyang** [comment]
> Actually, it is the opposite. Docker needs to be installed first as flannel reuses/runs over docker's network. If there is a problem after re-running the playbook, please open an issue with steps how to reproduce it, with logs, etc.

Without docker network, flanneld can be started and running normally. But if flanneld are failed to start, docker cannot be started. 
Open #1483 for this.


> The is reasonable. Not sure if the kubernetes-addons role checks if all addons are running. From my observations it seems it does not. So even if any of the nodes is not ready, it will get ready eventually. Once done, addons gets scheduled and running. Still, I agree with that. Can you open an issue for that?

open #1484 for this.

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
Based on [1], all ``restart-<role>.sh`` scripts need to set ``--extra-vars "[<service>-modified=true]+"`` so all tasks tagged with ``restart`` are actually run. 

[1] http://docs.ansible.com/ansible/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
Maybe it is worth breaking this PR out into smaller PRs. Like:

* refactor node, master and kubernetes roles
* refactor github-release installation type
* add distribution-rpm installation type
* introduce new playbooks and scripts

Though, at the end it is just a list of commits.

**Comment by ingvagabund** [comment]
@sttts

File: ansible/roles/kubernetes/tasks/main.yml [hunk 2]
```diff
@@ -1,36 +1,13 @@
 ---
-- include: fedora.yml
-  when: ansible_distribution == "Fedora"
+- name: Fedora | Remove docker window manager on F20
+  yum: pkg=docker state=absent
```
Comment (sttts) [inline ansible/roles/kubernetes/tasks/main.yml]: Are those ansible files explicitly only for Red Hat Linux variants?

File: ansible/roles/kubernetes/tasks/main.yml [hunk 3]
```diff
@@ -1,36 +1,13 @@
...
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version == "20"
```
Comment (sttts) [inline ansible/roles/kubernetes/tasks/main.yml]: This answers my own question.

File: ansible/roles/master/tasks/controller-manager-configure.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+- name: write the config file for the controller-manager
+  template: src=controller-manager.j2 dest={{ kube_config_dir }}/controller-manager
+
+- name: write the kubecfg (auth) file for controller-manager
```
Comment (sttts) [inline ansible/roles/master/tasks/controller-manager-configure.yml]: lower or update case names? Pretty mixed everywhere.

File: ansible/roles/master/tasks/controller-manager-start.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Enable controller-manager
+  service: name=kube-controller-manager enabled=yes state=started
+
+- name: Enable controller-manager
```
Comment (sttts) [inline ansible/roles/master/tasks/controller-manager-start.yml]: twice the same name?

File: ansible/roles/master/tasks/download_bins.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  set_fact:
+    kube_binaries: "{{ kube_master_binaries }} kubelet"
+  when: networking == 'opencontrail'
+
+- name: Download Kubernetes binaries
+  get_url:
+    url: "{{ kube_download_url_base }}/{{ item }}"
+    dest: "{{ kube_current_release_directory }}/{{ item }}"
+    mode: 0755
+    validate_certs: False
```
Comment (sttts) [inline ansible/roles/master/tasks/download_bins.yml]: Bad idea to load binaries from hosts without valid certs.

File: ansible/roles/master/tasks/retrieve-systemd-files.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+- name: Get Systemd Unit Files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/{{ item }}.service
```
Comment (sttts) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: Not a good idea to load unversioned files.

File: ansible/roles/master/tasks/retrieve-systemd-files.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    - kube-scheduler
+    - kubelet
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd Config Files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/environ/{{ item }}
```
Comment (sttts) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: again, unversioned

File: ansible/roles/master/tasks/retrieve-systemd-files.yml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    - controller-manager
+    - kubelet
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd Tmpfile from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/tmpfiles.d/kubernetes.conf
```
Comment (sttts) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: again, unversioned

File: ansible/roles/node/tasks/coreos.yml [hunk 10]
```diff
@@ -1,8 +1,4 @@
 ---
-- name: CoreOS | Force source_type to github
-  set_fact:
-    source_type: "github-release"
-
 - name: CoreOS | Get Systemd Unit Files from Kubernetes repository
   get_url:
     url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/{{ item }}.service
```
Comment (sttts) [inline ansible/roles/node/tasks/coreos.yml]: again, unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 11]
```diff
@@ -0,0 +1,46 @@
+---
+- name: Get Systemd Unit Files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/{{ item }}.service
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: again, unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
+- name: Get Systemd Unit Files from Kubernetes repository
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: many capital letters.

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    - kubelet
+    - kube-proxy
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd Config Files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/environ/{{ item }}
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: again, unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    - kubelet
+    - proxy
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd Tmpfile from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/tmpfiles.d/kubernetes.conf
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: again, unversioned

File: ansible/roles/master/tasks/controller-manager-start.yml (hunk 5 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/controller-manager-start.yml]: The first one needs to be removed.

File: ansible/README.md [hunk 15]
```diff
@@ -66,6 +66,41 @@ You can just setup certain parts instead of doing it all.
...
+Each component can be installed from various sources. For instance:
+
+* distribution packages
+* github release
+* kubernetes built from local source codes
+
+By default, every component (etcd, docker, kubernetes, etc.) is installed via distribution package manager.
+Currently, the following component types are supported:
+
+* ``etcd_source_type``: for ``etcd`` role
```
Comment (sttts) [inline ansible/README.md]: why double `\`` ?

File: ansible/roles/master/tasks/controller-manager-configure.yml (hunk 4 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/controller-manager-configure.yml]: That is a problem of every task. I will fix it. All names should be upper-cased.

File: ansible/README.md [hunk 16]
```diff
@@ -66,6 +66,41 @@ You can just setup certain parts instead of doing it all.
...
+* ``flannel_source_type``: for ``flannel`` role
+* ``kube_source_type``: for ``master`` and ``node`` roles
+* ``source_type``: for other roles (and components)
+
+To see a full list of available types, see corresponding role's default variables.
+
+#### kubernetes source type
```
Comment (sttts) [inline ansible/README.md]: capital?

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 13 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: Write, the config file should be taken from contrib repository itself.

File: ansible/roles/master/tasks/retrieve-systemd-files.yml (hunk 7 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: It fails under https://github.com/kubernetes/contrib/issues/931.

File: ansible/roles/master/tasks/download_bins.yml (hunk 6 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/download_bins.yml]: As currently all the binaries are downloaded directly from github.com/kubernetes/kubernetes there is no need for it. But yes, is should be validated by default.

Comment (ingvagabund) [inline ansible/roles/master/tasks/download_bins.yml]: https://github.com/kubernetes/contrib/issues/1544

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 12 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: addressed

File: ansible/roles/master/tasks/controller-manager-configure.yml (hunk 4 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/controller-manager-configure.yml]: Addressed up to some cases

File: ansible/roles/master/tasks/retrieve-systemd-files.yml (hunk 8 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: addressed

File: ansible/README.md (hunk 16 above)
Comment (ingvagabund) [inline ansible/README.md]: addressed

File: ansible/README.md (hunk 15 above)
Comment (ingvagabund) [inline ansible/README.md]: habit, updated to single `

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 12 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: addressed

**Comment by ingvagabund** [comment]
@sttts comments addressed, PTAL

File: ansible/README.md [hunk 17]
```diff
@@ -66,6 +66,41 @@ You can just setup certain parts instead of doing it all.
...
+#### Kubernetes source type
+
+Available types (see `kube_source_type` under `roles/kubernetes/defaults/main.yml`):
+
+* `packageManager`
+* `localBuild`
+* `github-release`
+* `distribution-rpm`
+
+In a case of a package manager, `kube-apiserver` binary is shipped with `cap_net_bind_service=ep` capability set.
```
Comment (sttts) [inline ansible/README.md]: In case of a package manager

Comment (sttts) [inline ansible/README.md]: the `...` binary

File: ansible/README.md [hunk 18]
```diff
@@ -66,6 +66,41 @@ You can just setup certain parts instead of doing it all.
...
+
+To see a full list of available types, see corresponding role's default variables.
+
+#### Kubernetes source type
+
+Available types (see `kube_source_type` under `roles/kubernetes/defaults/main.yml`):
+
+* `packageManager`
+* `localBuild`
+* `github-release`
```
Comment (sttts) [inline ansible/README.md]: CamelCase or camel-case?

File: ansible/README.md [hunk 19]
```diff
@@ -66,6 +66,41 @@ You can just setup certain parts instead of doing it all.
...
+* `distribution-rpm`
+
+In a case of a package manager, `kube-apiserver` binary is shipped with `cap_net_bind_service=ep` capability set.
+The capability allows the apiserver to listen on `443` port.
+In a case of `localBuild` and `github-release`, the capability is not set.
+In order for apiserver to listen on a secure port, change the port (see `kube_master_api_port` under `roles/kubernetes/defaults/main.yml`). For instance to listen on `6443`.
+
+In order to apply `distribution-rpm` type, location of an rpm must be specified.
```
Comment (sttts) [inline ansible/README.md]: the `...` type, the location

File: ansible/README.md [hunk 20]
```diff
@@ -74,6 +109,6 @@ By changing the `networking` variable in the `inventory/group_vars/all.yml` file
 
 ### Troubleshooting
 
-* When updating flannel to version ``0.5.5-7`` or higher on Fedora, ``/etc/sysconfig/flannel`` configuration file (if changed) must be updated to reflect renamed systemd environment variables.
+* When updating flannel to version `0.5.5-7` or higher on Fedora, `/etc/sysconfig/flannel` configuration file (if changed) must be updated to reflect renamed systemd environment variables.
```
Comment (sttts) [inline ansible/README.md]: the `...` file

File: ansible/roles/master/tasks/kubelet-configure.yml [hunk 21]
```diff
@@ -0,0 +1,8 @@
+---
+- name: Write the config files for kubelet
+  template: src=kubelet.j2 dest={{ kube_config_dir }}/kubelet
```
Comment (sttts) [inline ansible/roles/master/tasks/kubelet-configure.yml]: Why no consistent naming? kubelet vs. kubectl.kubeconfig

File: ansible/roles/master/tasks/ubuntu-14.04.yml [hunk 22]
```diff
@@ -1,8 +1,4 @@
 ---
-- name: Ubuntu | Force source_type to github
-  set_fact:
-    source_type: "github-release"
-
 - name: Ubuntu | Get Upstart Config Files from Kubernetes repository
   get_url:
     url=https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/ubuntu/master/init_conf/{{ item }}.conf
```
Comment (sttts) [inline ansible/roles/master/tasks/ubuntu-14.04.yml]: unversioned. not good.

File: ansible/roles/node/tasks/fluentd-install.yml [hunk 23]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Install fluentd pod into each node
+  get_url:
+    url=https://raw.githubusercontent.com/GoogleCloudPlatform/kubernetes/master/cluster/saltbase/salt/fluentd-es/fluentd-es.yaml
```
Comment (sttts) [inline ansible/roles/node/tasks/fluentd-install.yml]: unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 24]
```diff
@@ -0,0 +1,46 @@
+---
+- name: Get Systemd unit files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/{{ item }}.service
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+    - kubelet
+    - kube-proxy
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd config files from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/environ/{{ item }}
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: unversioned

File: ansible/roles/node/tasks/retrieve-systemd-files.yml [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+    - kubelet
+    - proxy
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Get Systemd tmpfile from Kubernetes repository
+  get_url:
+    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/tmpfiles.d/kubernetes.conf
```
Comment (sttts) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: unversioned

File: ansible/scripts/restart-master.sh [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+. ./init.sh
+
+inventory=${INVENTORY:-${INVENTORY_DIR}/inventory}
+ansible-playbook -i ${inventory} ${PLAYBOOKS_DIR}/deploy-master.yml \
+		--tags "restart" \
```
Comment (sttts) [inline ansible/scripts/restart-master.sh]: huge indention. 2 spaces are enough.

**Comment by sttts** [comment]
@ingvagabund main comment: use ansible host files from the repo instead of remote downloads (without versions and partly without certificate checks).

File: ansible/roles/master/tasks/kubelet-configure.yml (hunk 21 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/kubelet-configure.yml]: ``kubelet.kubeconfig`` would have different meaning. ``kubelet`` is configurable for ``kubelet`` component, ``kubectl.kubeconfig`` is configuration for connecting to ``apiserver``.

File: ansible/roles/master/tasks/ubuntu-14.04.yml (hunk 22 above)
Comment (ingvagabund) [inline ansible/roles/master/tasks/ubuntu-14.04.yml]: Will be fixed as part of #931

File: ansible/roles/node/tasks/fluentd-install.yml (hunk 23 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/fluentd-install.yml]: Will be fixed as part of #931

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 24 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: Will be fixed as part of #931

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 25 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: Will be fixed as part of #931

File: ansible/roles/node/tasks/retrieve-systemd-files.yml (hunk 26 above)
Comment (ingvagabund) [inline ansible/roles/node/tasks/retrieve-systemd-files.yml]: Will be fixed as part of #931

File: ansible/scripts/restart-master.sh (hunk 27 above)
Comment (ingvagabund) [inline ansible/scripts/restart-master.sh]: I wanted to align all options as close to each other as possible. What about

```
ansible-playbook \
  -i ${inventory} ${PLAYBOOKS_DIR}/deploy-master.yml \
  --tags "restart" \
  --extra-vars "apiserver_modified=true controller_manager_modified=true scheduler_modified=true kubelet_modified=true" \
  "$@"
```

?

File: ansible/README.md (hunk 20 above)
Comment (ingvagabund) [inline ansible/README.md]: fixed

File: ansible/README.md (hunk 19 above)
Comment (ingvagabund) [inline ansible/README.md]: fixed

File: ansible/README.md (hunk 17 above)
Comment (ingvagabund) [inline ansible/README.md]: fixed

File: ansible/README.md (hunk 18 above)
Comment (ingvagabund) [inline ansible/README.md]: currently both, ``camel-case`` looks better. Issue filled at https://github.com/kubernetes/contrib/issues/1600.

**Comment by ingvagabund** [comment]
> @ingvagabund main comment: use ansible host files from the repo instead of remote downloads (without versions and partly without certificate checks).

Agree, the ansible playbook should be as much self-standing as possible. ``kubernetes`` like roles are not the only roles that needs the improvement. Let's put it as a separate PR so all changes are made at once.

**Comment by ingvagabund** [comment]
@sttts PTAL

**Comment by sttts** [comment]
With the outlook of #931, the remaining changes lgtm, as far as I can judge.

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by jasonbrooks** [comment]
@ingvagabund I'm having issues using this with centos/fedora/centosatomic with vagrant. I'm getting the error:

```
TASK [etcd : Write etcd config file] *******************************************
fatal: [kube-master-1]: FAILED! => {"changed": false, "failed": true, "msg": "AnsibleUndefinedVariable: 'networking' is undefined"}
```

My networking is set to the default flannel value. I've poked at it for a while, but I can't tell why it's going wrong... The `etcd.conf.j2` has some `if networking == "contiv"` elements, but these have been here for a while. Maybe something to do with the reorganization?

**Comment by ingvagabund** [comment]
@jasonbrooks it is related to this issue https://github.com/kubernetes/contrib/pull/933. Short fix is to run

```
ln -fs ../../../../../inventory/group_vars .vagrant/provisioners/ansible/inventory
```

in ansible/vagrant directory.

**Pull request merged**